doozer/doozerlib/cli/release_gen_payload.py

```diff
@@ -761,25 +767,50 @@ async def sync_payloads(self):
             False: dict()
         }
 
-        # Ensure that all payload images have been mirrored before updating
-        # the imagestream. Otherwise, the imagestream will fail to import the
-        # image.
+        # Filter out embargoed builds to mirror to public quay
+        filtered_payload_entries_for_arch = {}
+        for arch, payload_entries in self.payload_entries_for_arch.items():
+            filtered_payload_entries = {}
+            for payload_tag_name, payload_entry in payload_entries.items():
+                if payload_entry.rhcos_build is None and payload_entry.build_inspector.is_under_embargo():
+                    # (note RHCOS builds can't be embargoes for now)
+                    if self.runtime.assembly_type is not AssemblyTypes.STANDARD:
+                        # Unless releasing, we do not want embargoed builds synced to public quay
+                        continue
+                    if self.embargo_permit_ack and self.runtime.contains_embargo:
+                        # consciously releasing private content
+                        self.includes_embargo = True
+                    else:
+                        raise IOError(f"Embargoed build detected for payload tag {payload_tag_name} with "
```

Comment:
Iterate through all results before throwing the error. Imagine if there were 2 embargoed images in the payload: one that was ready to be unembargoed and one that was not. You'd want the artist to know about both before making the decision to permit the unembargo.

Reply:
easy if it's just adding an `AssemblyIssue`. but this may require a little rearrangement; this sub doesn't currently contribute to assembly issues, and i don't recall where we actually check for viability. we want to make sure that's given a chance to evaluate and stop issues before we've actually copied any content out.